pkg/slayers/path/scion/raw_spec.gobra

```diff
@@ -575,42 +575,76 @@ pure func (s *Raw) absPkt(dp io.DataPlaneSpec, raw []byte) (res io.IO_pkt2) {
 		})
 }
 
+// In the future, this should probably use AbsMetaHdr as
+// the return type.
+ghost
+requires MetaLen <= len(raw)
+requires acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), R56) 
+decreases
+pure func RawBytesToMetaHdr(raw []byte) MetaHdr {
+	return unfolding acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), R56) in
+		let hdr := binary.BigEndian.Uint32(raw[:MetaLen])          in
+		DecodedFrom(hdr)
+}
+
+// In the future, this should probably use AbsBase as
+// the return type.
+ghost
+requires MetaLen <= len(raw)
+requires acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), R56)
+decreases
+pure func RawBytesToBase(raw []byte) Base {
+	return let metaHdr := RawBytesToMetaHdr(raw) in
+		let seg1 := int(metaHdr.SegLen[0])       in
+		let seg2 := int(metaHdr.SegLen[1])       in
+		let seg3 := int(metaHdr.SegLen[2])       in
+		Base{metaHdr, NumInfoFields(seg1, seg2, seg3), seg1 + seg2 + seg3}
+}
+
 ghost
 opaque
 requires acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), R56)
 decreases
 pure func validPktMetaHdr(raw []byte) bool {
 	return MetaLen <= len(raw)  &&
-		unfolding acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), R56) in
-		let hdr := binary.BigEndian.Uint32(raw[:MetaLen]) in
-		let metaHdr := DecodedFrom(hdr) in
-		let seg1 := int(metaHdr.SegLen[0]) in
-		let seg2 := int(metaHdr.SegLen[1]) in
-		let seg3 := int(metaHdr.SegLen[2]) in
-		let base := Base{metaHdr,
-			NumInfoFields(seg1, seg2, seg3),
-			seg1 + seg2 + seg3} in
-		metaHdr.InBounds() &&
-		0 < metaHdr.SegLen[0] &&
-		base.ValidCurrIdxsSpec() &&
+		let metaHdr := RawBytesToMetaHdr(raw) in
+		let seg1 := int(metaHdr.SegLen[0])    in
+		let seg2 := int(metaHdr.SegLen[1])    in
+		let seg3 := int(metaHdr.SegLen[2])    in
+		let base := RawBytesToBase(raw)       in
+		0 < metaHdr.SegLen[0]           &&
+		base.ValidCurrIdxsSpec()        &&
 		pktLen(seg1, seg2, seg3, 0) <= len(raw)
```

Comment:
We should use these simplifications in 'validPktMetaHdr` in ìo-spec.gobra` as well.